src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts

```diff
@@ -107,6 +154,127 @@ export class ExerciseChatWidgetComponent implements OnInit, OnDestroy, AfterView
         this.dialog.closeAll();
     }
 
+    acceptPermission() {
+        this.accountService.identity().then((user: User) => {
+            if (typeof user!.login === 'string') {
```

Comment:
TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.   Type 'undefined' is not assignable to type 'string'.

Otherwise we can not store in localStorage